### PR TITLE
[Main2Main] Upgrade vllm commit to 0123

### PR DESCRIPTION
**Pull request by Meihan-chen**
### What this PR does / why we need it?
1. ✅ Upgrade vllm commit to: 0115 (8471b27df97c3eb79f891802fc0e858f8f7ac6a0)
Modify import paths due to the refactors：
https://github.com/vllm-project/vllm/pull/32245
https://github.com/vllm-project/vllm/pull/32060
Test result: https://github.com/vllm-project/vllm-ascend/actions/runs/21034239336/job/60490156965?pr=5913
2.  ✅Upgrade vllm commit to: 0119 (9a1f16da1e423ede2c2f52a9850cbfbb39cefe96)
Fix `WorkerProc.__init__() missing 1 required positional argument: 'is_driver_worker'` due to https://github.com/vllm-project/vllm/pull/28506
Test result: https://github.com/vllm-project/vllm-ascend/actions/runs/21156263050/job/60841668755?5569
3.  ✅Upgrade vllm commit to: 0120(148117ea2e689cd43df4be6892671a17cdae5833)
     1. Add `skip_compiled` param in `set_forward_context` due to https://github.com/vllm-project/vllm/pull/30385 
     2. Modify  `tests/ut/spec_decode/test_eagle_proposer.py` due to https://github.com/vllm-project/vllm/pull/24322
change `self.max_num_tokens = vllm_config.scheduler_config.max_num_batched_tokens + max_batch_size`
     3. Modify UT import paths due to the refactors：https://github.com/vllm-project/vllm/pull/32060
 Test result: https://github.com/vllm-project/vllm-ascend/actions/runs/21204851770/job/60999046946
 4.  ✅Upgrade vllm commit to: 0121(f23fb5a7c1b61350c5c40ca1115d3bf8cf2b8cc9)
      1. vLLM switched `uses_mrope` from target to draft model config, making `positions`/`mrope_positions` mutually exclusive, breaking vllm-ascend's direct self.positions access and tests missing `draft_model_config.uses_mrope`. https://github.com/vllm-project/vllm/pull/32048
      2. Moved bs_to_padded_graph_size from CompilationConfig to CudagraphDispatcher due to the refactor https://github.com/vllm-project/vllm/pull/30143
      3. Remove unused `maybe_setup_kv_connector` due to  https://github.com/vllm-project/vllm/pull/32077
Test result: https://github.com/vllm-project/vllm-ascend/actions/runs/21217728738/job/61043738834
 6.  ✅Upgrade vllm commit to: 0122(8ebf271bb6d1e7e9b1a55be73d755ef1a57dbbe5)
      Updating FusedMoEParallelConfig (added enable_eplb) and FusedMoEConfig due to https://github.com/vllm-project/vllm/pull/32414
Test result: https://github.com/vllm-project/vllm-ascend/actions/runs/21249922546/job/61148613054
 8.  ✅Upgrade vllm commit to: 0123(dc917cceb877dfd13f98c538c4c96158047d98bd)
      Setting temperature=0.0 due to the removal of the default temperature value in https://github.com/vllm-project/vllm/pull/32723
Test result:  https://github.com/vllm-project/vllm-ascend/actions/runs/21280796875
### How was this patch tested?

- vLLM version: v0.14.0
- vLLM main: https://github.com/vllm-project/vllm/commit/d68209402ddab3f54a09bc1f4de9a9495a283b60

### CODE REVIEW

**Comment by Meihan-chen** [comment]
All checks have passed in https://github.com/vllm-project/vllm-ascend/actions/runs/21348540836/job/61440604090 
https://github.com/vllm-project/vllm-ascend/actions/runs/21353410481/job/61455937068

File: tests/e2e/multicard/2-cards/test_aclgraph_capture_replay.py [hunk 1]
```diff
@@ -206,7 +204,8 @@ def test_models_aclgraph_capture_replay_metrics_dp2(
     # 2. Generation steps (max_tokens)
     # 3. Final step (likely EOS/idle step), no replay here
     total_steps = max_tokens + 1  # this includes the 1 and 2 above
-    expected_exec_model = (total_steps + 1) * dp_size
+    # vllm default enables Async scheduler, this will take 1 more steps
+    expected_exec_model = (total_steps + 1 + 1) * dp_size
```
Comment (wangxiyuan) [inline tests/e2e/multicard/2-cards/test_aclgraph_capture_replay.py:208]: 0.14.0 enable async scheduler already. why it's fine before?

File: tests/e2e/multicard/2-cards/test_qwen3_performance.py [hunk 2]
```diff
@@ -42,6 +42,7 @@
     "random_input_len": 128,
     "max_concurrency": 40,
     "random_output_len": 100,
+    "temperature": 0.0,
```
Comment (wangxiyuan) [inline tests/e2e/multicard/2-cards/test_qwen3_performance.py:45]: I think we should add it to benchamark module? @wxsIcey

File: vllm_ascend/patch/platform/patch_multiproc_executor.py [hunk 3]
```diff
@@ -77,6 +70,7 @@ def _init_executor(self) -> None:
             global_start_rank = self.local_world_size * self.parallel_config.node_rank_within_dp
             for local_rank in range(self.local_world_size):
                 global_rank = global_start_rank + local_rank
+                is_driver_worker = self._is_driver_worker(global_rank)
```
Comment (wangxiyuan) [inline vllm_ascend/patch/platform/patch_multiproc_executor.py:73]: is this change work with 0.14.0?

File: vllm_ascend/worker/v2/sample/penalties.py [hunk 4]
```diff
@@ -20,7 +20,7 @@
 
 import torch
 from vllm.triton_utils import tl, triton
-from vllm.v1.worker.gpu.sample.metadata import SamplingMetadata
+from vllm.v1.sample.metadata import SamplingMetadata
```
Comment (wangxiyuan) [inline vllm_ascend/worker/v2/sample/penalties.py:23]: work with v0.14.0?

File: tests/e2e/multicard/2-cards/test_aclgraph_capture_replay.py (hunk 1 above)
Comment (Meihan-chen) [reply tests/e2e/multicard/2-cards/test_aclgraph_capture_replay.py:208]: To adapt to previous upgrades, it was set to false in https://github.com/vllm-project/vllm-ascend/pull/6169/changes/BASE..181a3a373cc9e8f37908c3d14ae83d60dd8b4f6e#diff-0e3dd1c713072facdea0241f1b908c2121957baf6f0400fd71f9b8224f2c98d1L112； this config should remov when the VLLM version is >= 0.14.0.

File: vllm_ascend/patch/platform/patch_multiproc_executor.py (hunk 3 above)
Comment (Meihan-chen) [reply vllm_ascend/patch/platform/patch_multiproc_executor.py:73]: is_driver_worker is vllm-ascend's own make_worker_process method which always accepts this parameter, while AscendWorkerProc conditionally passes it to vllm's native WorkerProc.__init__ which only accepts it in newer versions，so only  https://github.com/vllm-project/vllm-ascend/pull/6169/changes/BASE..181a3a373cc9e8f37908c3d14ae83d60dd8b4f6e#diff-b25d0746a873edc4bc07a475dd926377fdda5e51a437fc329123eea9202cfdaaR181 needs version checking.
